### PR TITLE
fix: [CCM-8597]: Perspective filter sorting on filterKey

### PR DESCRIPTION
**Pull request by divyanshuchaubisa**
## Describe your changes
* Applied sort criteria on queries to fetch the perspective filter results.
* Applied list sort in case of AWS account ids after getting the final list from bigQuery and mongoDB.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37442)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 1 Issues 
* Anti-pattern 1 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37442)

File: 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/ViewsBillingServiceImpl.java [hunk 1]
```diff
@@ -284,8 +285,11 @@ public List<String> getFilterValueStatsNg(BigQuery bigQuery, List<QLCEViewFilter
       Thread.currentThread().interrupt();
       return null;
     }
-    return costCategoriesPostFetchResponseUpdate(getFilterValuesData(queryParams.getAccountId(), viewsQueryMetadata,
-                                                     result, idFilters, cloudProviderTableName.contains(CLUSTER_TABLE)),
+
+    return costCategoriesPostFetchResponseUpdate(
+        awsAccountFieldHelper.spiltAndSortAWSAccountIdListBasedOnAccountName(
```
Comment (sdsingh07) [inline 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/ViewsBillingServiceImpl.java:290]: We have to call `spiltAndSortAWSAccountIdListBasedOnAccountName` function only if idFilter has AWS account, not for all values. We can put this function in `getFilterValuesData` function where we are already checking if data is filtered by AWS account.

File: 450-ce-views/src/main/java/io/harness/ccm/views/helper/AwsAccountFieldHelper.java [hunk 2]
```diff
@@ -173,4 +174,20 @@ public static List<QLCEViewRule> removeAccountNameFromAWSAccountRuleFilter(final
     });
     return updatedRuleFilters;
   }
+
+  public List<String> spiltAndSortAWSAccountIdListBasedOnAccountName(final List<String> values) {
+    List<String> accountIdsWithNames = new ArrayList<>();
+    List<String> accountIdsWithoutNames = new ArrayList<>();
+    for (String value : values) {
+      if (Character.isDigit(value.charAt(0))) {
```
Comment (sdsingh07) [inline 450-ce-views/src/main/java/io/harness/ccm/views/helper/AwsAccountFieldHelper.java:182]: Account name can start with digit. We need to change this logic.

https://47lining.github.io/nucleator/documentation/Sign-Up-for-New-AWS-Account-using-Nucleator-Naming-Conventions_20545589.html#SignUpforNewAWSAccountusingNucleatorNamingConventions-AWSAccountName

**Review by sdsingh07 — COMMENTED** [review]
> Reviewed 1 of 2 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37442)* status: 0 of 1 approvals obtained.

**Review by sdsingh07 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37442#-:-NBqN71XEKucOD7mVmPt:bnfp4nl)* status: :shipit: complete! ✓

**Comment by divyanshuchaubisa** [comment]
trigger ti0, ti1, ti2

**Pull request merged**